api/graphql/schema.graphql

```diff
@@ -1674,10 +1693,25 @@ input UpdateScreeningQuestionResponsesHasMany {
   update: [UpdateScreeningQuestionResponseInput!]
   delete: [ID!]
 }
+
+input UpdateExperiencesCriteriaBelongsToMany {
+  connect: [ID!]
+  sync: [ID!]
+  syncWithoutDetaching: [ID!]
+  disconnect: [ID!]
+}
+
 input UpdateApplicationInput {
   # can only connect a step, not disconnect or sync
   insertSubmittedStep: ApplicationStep
   screeningQuestionResponses: UpdateScreeningQuestionResponsesHasMany
+  minimumCriteria: ApplicationCriteria @rename(attribute: "minimum_criteria")
+  # experiences for meeting minimum criteria
+  awardExperiencesCriteria: UpdateExperiencesCriteriaBelongsToMany
+  communityExperiencesCriteria: UpdateExperiencesCriteriaBelongsToMany
+  educationExperiencesCriteria: UpdateExperiencesCriteriaBelongsToMany
+  personalExperiencesCriteria: UpdateExperiencesCriteriaBelongsToMany
+  workExperiencesCriteria: UpdateExperiencesCriteriaBelongsToMany
```

Comment:
Hmm... Is there a way to write this with a single, generic experiences field? If not, I may have to take back some of my above comments.

Reply:
Not sure, given this was another case of polymorphism, relating the pool candidates table to one of the five experiences tables, I followed suit setting things up ala skills to experiences.

Reply:
Hm. Well, I just did some experimenting. We could move the work of sorting experience types to the backend, with a mutator:
```
#schema
input ExperienceIdInput {
  id: ID!
  typename: String!
}

input UpdateApplicationInput {
  # can only connect a step, not disconnect or sync
  insertSubmittedStep: ApplicationStep
  screeningQuestionResponses: UpdateScreeningQuestionResponsesHasMany
  minimumCriteria: ApplicationCriteria @rename(attribute: "minimum_criteria")
  syncCriteriaExperiences: [ExperienceIdInput!]
}

#PoolCandidate.php
  public function setSyncCriteriaExperiencesAttribute($experiences)
  {
      $workIds = collect($experiences)->filter(fn($item) => $item['typename'] == 'WorkExperience')->pluck('id');
      $educationIds = collect($experiences)->filter(fn($item) => $item['typename'] == 'EducationExperience')->pluck('id');
      $awardIds = collect($experiences)->filter(fn($item) => $item['typename'] == 'AwardExperience')->pluck('id');
      $personalIds = collect($experiences)->filter(fn($item) => $item['typename'] == 'PersonalExperience')->pluck('id');
      $communityIds = collect($experiences)->filter(fn($item) => $item['typename'] == 'CommunityExperience')->pluck('id');

      $this->awardExperiencesCriteria()->sync($awardIds);
      $this->communityExperiencesCriteria()->sync($communityIds);
      $this->educationExperiencesCriteria()->sync($educationIds);
      $this->personalExperiencesCriteria()->sync($personalIds);
      $this->workExperiencesCriteria()->sync($workIds);
  }
```
But I don't know if that will really be easier for the frontend. I could use a second opinion. @petertgiles could you let me know what you think of structuring this particular mutation input?

Reply:
I'm doubtful it will make the frontend code much easier.  We already have a control on the page to sort experiences and only show one type so picking the right field for them in the mutation doesn't set us back much.  Our schema already has separate types for all the experience types so I don't think we're leaking anything by having separate fields for each type here.